contracts/dao/voting/vesting-lp-vault/src/contract.rs

```diff
@@ -0,0 +1,343 @@
+        QueryMsg::GetConfig {} => query_config(deps),
+        QueryMsg::ListBonders { start_after, limit } => {
+            query_list_bonders(deps, start_after, limit, None)
+        }
+        QueryMsg::BondingStatus { height, address } => {
+            to_binary(&query_bonding_status(deps, env, height, address)?)
+        }
+    }
+}
+
+pub fn query_voting_power_at_height(
+    deps: Deps,
+    env: Env,
+    address: String,
+    height: Option<u64>,
+) -> StdResult<VotingPowerAtHeightResponse> {
+    let config = CONFIG.load(deps.storage)?;
+    let height = height.unwrap_or(env.block.height);
+
+    let query_msg = VestingLpQueryMsg::UnclaimedAmountAtHeight { address, height };
+    let atom_power = get_voting_power(
+        deps,
+        &config.vesting_lp_contract,
+        &config.atom_oracle_contract,
+        &query_msg,
+        height,
+    )?;
+    let usdc_power = get_voting_power(
+        deps,
+        &config.vesting_lp_contract,
+        &config.usdc_oracle_contract,
+        &query_msg,
+        height,
+    )?;
+
+    Ok(VotingPowerAtHeightResponse {
+        power: (atom_power + usdc_power).numerator().try_into()?,
+        height,
+    })
+}
+
+pub fn query_total_power_at_height(
+    deps: Deps,
+    env: Env,
+    height: Option<u64>,
+) -> StdResult<TotalPowerAtHeightResponse> {
+    let config = CONFIG.load(deps.storage)?;
+    let height = height.unwrap_or(env.block.height);
+
+    let query_msg = VestingLpQueryMsg::UnclaimedTotalAmountAtHeight { height };
+    let atom_power = get_voting_power(
+        deps,
+        &config.vesting_lp_contract,
+        &config.atom_oracle_contract,
+        &query_msg,
+        height,
+    )?;
+    let usdc_power = get_voting_power(
+        deps,
+        &config.vesting_lp_contract,
+        &config.usdc_oracle_contract,
+        &query_msg,
+        height,
+    )?;
+
+    Ok(TotalPowerAtHeightResponse {
+        power: (atom_power + usdc_power).numerator().try_into()?,
+        height,
+    })
+}
+
+pub fn query_info(deps: Deps) -> StdResult<Binary> {
+    let info = cw2::get_contract_version(deps.storage)?;
+    to_binary(&cwd_interface::voting::InfoResponse { info })
+}
+
+pub fn query_dao(deps: Deps) -> StdResult<Binary> {
+    let dao = DAO.load(deps.storage)?;
+    to_binary(&dao)
+}
+
+pub fn query_name(deps: Deps) -> StdResult<Binary> {
+    let config = CONFIG.load(deps.storage)?;
+    to_binary(&config.name)
+}
+
+pub fn query_description(deps: Deps) -> StdResult<Binary> {
+    let config = CONFIG.load(deps.storage)?;
+    to_binary(&config.description)
+}
+
+pub fn query_config(deps: Deps) -> StdResult<Binary> {
+    let config = CONFIG.load(deps.storage)?;
+    to_binary(&config)
+}
+
+pub fn query_list_bonders(
+    deps: Deps,
+    start_after: Option<String>,
+    limit: Option<u32>,
+    order_by: Option<OrderBy>,
+) -> StdResult<Binary> {
+    // TODO: this method does not adjust LP tokens amount to their corresponding voting power
```

Comment:
what is this TODO about? I believe it's an expected behaviour

Reply:
uhh it is not. the amount of LP tokens doesn't correspond to voting power. Nevertheless, I wasn't expected to implement this method, it is just a draft for future implementation.

Reply:
maybe it makes sense to modify the response structure here and have two different fields for usdc and atom. but it depends on the business needs I believe because this query seems to be more for users, not for other contracts

Reply:
CC @swelf19 @ratik

Reply:
have no clue why there should be usdc/atom lb balances, does it relate to dao?

Reply:
because we have a voting power specific queries which return voting power, but this one is about bond/unbond which is about concrete funds put to/taken from the contract. we also have a BondingStatus query which also returns more of balance than voting power:

```rust
pub struct BondingStatusResponse {
    pub bonding_enabled: bool,
    pub unbondable_abount: Uint128,
    pub height: u64,
}
```

Reply:
there are some of options i can see:
1) Disable the query at all. Return Err or some meaningful response. Because contract does not accumulate any assests. It just queries vesting-lp balance to calculate the voting power.
2) Modify the response - return the amount of lp tokens and the power they represent.
Both options are appropriate from my point of view

furthermore. I can see you use only one vesting-lp contract. This is incorrect. Single vesting-lp represent singlle lptoken denom. One vesting-lp for ATOM liquidy pool and one for USDC liquidity pool. You either need to configure second vesting-lp or refactor the contract to work with a single denom. In second case we just need to setup two vesting valults

Reply:
okay make sense. As currently we don't have any need to display lp balance here, from my point of view we can leave it as is for now

Reply:
> You either need to configure second vesting-lp or refactor the contract to work with a single denom. In second case we just need to setup two vesting valults

What, I thought it is okay for a contract to add together voting powers represented by both denoms? :thinking:

Reply:
ignore mine. agree with @swelf19 as he addressed it great

Reply:
> What, I thought it is okay for a contract to add together voting powers represented by both denoms? thinking
its kinda ok for your contract. But you query vesting_lp contract. We do not have single vesting_lp. We have vesting_lp_atom + vesting_lp_usdc, you have to query appropriate contract for every denom.

Reply:
> 1. Disable the query at all. Return Err or some meaningful response. Because contract does not accumulate any assests. It just queries vesting-lp balance to calculate the voting power.

Wow, interesting point of view. I personally like the first solution more because there are no bonders indeed since bonding is disabled

Reply:
thank you all! I have disabled this query